.github/workflows/km-valgrind.yaml

```diff
@@ -0,0 +1,196 @@
+  start-runners:
+    name: Start cloud runners
+    runs-on: ubuntu-latest
+    outputs:
+      run-ons: ${{ steps.start-cloud-runner.outputs.run-ons }}
+    steps:
+      - name: Get public Keys
+        uses: actions/checkout@v2
+
+      - name: Start Cloud runners
+        id: start-cloud-runner
+        uses: kontainapp/cloud-github-runner@v3.1
+        with:
+          mode: start
+          github-token: ${{ secrets.GH_TOKEN }}
+          runner-user: kontain
+          subnet: "10.0.0.0"
+          public-keys-dir: ${{ github.workspace }}/cloud/ssh
+          ec2-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          ec2-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+          ec2-region: "us-east-2"
+          ec2-image: "L0BaseImage"
+          ec2-instance-type: "m5.xlarge"
+          ec2-vpc-name: "github-runner"
+          az-subscription-id: ${{ secrets.SP_SUBSCRIPTION_ID }}
+          az-client-id: ${{ secrets.SP_APPID }}
+          az-secret: ${{ secrets.SP_PASSWORD }}
+          az-tenant-id: ${{ secrets.SP_TENANT }}
+          az-image: "auto-github-runner:L0BaseImage"
+          az-location: "westus"
+          az-vm-size: "Standard_D4s_v5"
+
+  km-build:
+    name: Build KM, push test image
+    runs-on: ubuntu-20.04
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          submodules: true
+
+      - run: make -C cloud/azure login-cli
+
+      - name: Prepare KM build env
+        run: make -C tests pull-buildenv-image .buildenv-local-lib
+
+      - name: Build KM and tests using buildenv image
+        run: make -j withdocker RUN_IN_CI=1 VALGRIND=1
+
+      - name: Build and push KM testenv image
+        run: make -C tests testenv-image push-testenv-image
+
+  km-test:
+    name: KM, KVM on Azure
+    runs-on: ${{ fromJSON(needs.start-runners.outputs.run-ons)['azure'] }}
+    needs: [start-runners, km-build]
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          submodules: true
+
+      - run: make -C cloud/azure login-cli
+
+      - run: |
+          echo 'KERNEL=="kvm", GROUP="kvm", MODE="0666"' > /tmp/rules
+          sudo mv /tmp/rules /etc/udev/rules.d/99-perm.rules
+          sudo udevadm control --reload-rules && sudo udevadm trigger
+
+      - name: Pull testenv images
+        run: make -C tests pull-testenv-image
+
+      - name: KM with KVM Test - Azure
+        run: make -C tests test-withdocker DOCKER_INTERACTIVE= DOCKER_RUN_CLEANUP= VALGRIND=1
+        timeout-minutes: 60
+
+  kkm-test-vms:
+    name: KM tests, KKM AWS
+    needs: [start-runners, km-build]
+    runs-on: ${{ fromJSON(needs.start-runners.outputs.run-ons)['ec2'] }}
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          submodules: true
+
+      - run: make -C cloud/azure login-cli
+
+      - name: KKM and KKM tests
+        run: |
+          make -C kkm/kkm VALGRIND=1 && make -C kkm/test_kkm
```

Comment:
Does the kkm build really need VALGRIND=1?

Reply:
Yes. valgrind doesn't really follow the PIE logic. Normally PIE gets loaded high in the memory, leaving lower addresses free for us to use. This is how we get the addresses in guest and km equal. However valgrind loads stuf right there, creating clashes. So for valgrind purposes we build km with 16TB shift, as we used to. KKM needs to be aware of that address range, which is simply hard coded for now with #define. Hence this.

We've discussed this with Srini, he suggested an `ioctl()` from km to kkm to notify kkm of the address range. When that's done this will be unnecessary.